pkg/utils/utils.go

```diff
@@ -136,10 +137,13 @@ func GetDateThreshold(date string) (time.Time, error) {
 }
 
 // FindChartURL will return the chart url
-func FindChartURL(name string, version string, index *helmRepo.IndexFile) (string, error) {
+func FindChartURL(name string, version string, index *helmRepo.IndexFile, sourceUrl string) (string, error) {
 	chart := findChartByVersion(index.Entries[name], version)
 	if chart != nil {
```

Comment:
You should probably also check if the `URLs` array is not empty to avoid an out-of-index error.

Reply:
```suggestion
	if chart != nil && len(chart.URLs) > 0 {
```

Reply:
good point